lib/widgets/inbox.dart

```diff
@@ -0,0 +1,518 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_color_models/flutter_color_models.dart';
+
+import '../api/model/model.dart';
+import '../model/narrow.dart';
+import '../model/recent_dm_conversations.dart';
+import '../model/unreads.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'sticky_header.dart';
+import 'store.dart';
+import 'colors.dart';
+import 'text.dart';
+import 'unread_count_badge.dart';
+
+class InboxPage extends StatefulWidget {
+  const InboxPage({super.key});
+
+  static Route<void> buildRoute({required BuildContext context}) {
+    return MaterialAccountWidgetRoute(context: context,
+      page: const InboxPage());
+  }
+
+  @override
+  State<InboxPage> createState() => InboxPageState();
+}
+
+class InboxPageState extends State<InboxPage> with PerAccountStoreAwareStateMixin<InboxPage> {
```

Comment:
Sure, it can be `@visibleForTesting`. If I make it private, by prefixing with `_`, I get analysis feedback saying "Invalid use of a private type in a public API" in several places. Such as:

```dart
class AllDmsHeaderItem extends StatelessWidget {
  const AllDmsHeaderItem({
    super.key,
    required this.count,
    required this.collapsed,
    required this.pageState, // <-- here
  });

  final int count;
  final bool collapsed;
  final _InboxPageState pageState; // <-- here
```

Reply:
Ah, I guess that points toward this other comment then :slightly_smiling_face: : https://github.com/zulip/zulip-flutter/pull/381#discussion_r1391752223